pkg/clusteragent/admission/patch/rc_provider.go

```diff
@@ -83,6 +87,34 @@ func (rcp *remoteConfigProvider) process(update map[string]state.APMTracingConfi
 		}
 		if ch, found := rcp.subscribers[req.K8sTarget.Kind]; found {
 			valid++
+			env := ""
```

Comment:
I would break this entire block of new code into its own function, that builds a `*telemetry.ApmRemoteConfigEvent` and then pass the output of that function into `rcp.telemetryCollector.SendEvent`.